ui/app/AppLayouts/Chat/panels/communities/CommunityMintTokensSettingsPanel.qml

```diff
@@ -252,22 +253,69 @@ SettingsPageLayout {
                                   StackView.Immediate)
             }
         }
     }
+
+    Component {
+        id: collectibleView
+
+        CommunityCollectibleView {
+            id: view
+
+            property int index // TODO: Update it to key when model has role key implemented
+
+            deployState: instantiator.object ? instantiator.object.deployState : CommunityCollectibleView.DeployState.Failed
+            name: instantiator.object ? instantiator.object.name : ""
+            artworkSource: instantiator.object ? instantiator.object.artworkSource : ""
+            symbol: instantiator.object ? instantiator.object.symbol : ""
+            description: instantiator.object ? instantiator.object.description : ""
+            supplyAmount: instantiator.object ? instantiator.object.supplyAmount : 0
+            infiniteSupply: instantiator.object ? instantiator.object.infiniteSupply : true
+            transferable: instantiator.object ? instantiator.object.transferable : false
+            selfDestruct: instantiator.object ? instantiator.object.selfDestruct : true
+            chainId: instantiator.object ? instantiator.object.chainId : -1
+            chainName: instantiator.object ? instantiator.object.chainName : ""
+            chainIcon: instantiator.object ? instantiator.object.chainIcon : ""
+            accountName: instantiator.object ? instantiator.object.accountName : ""
+            viewWidth: root.viewWidth
+            holdersModel: root.holdersModel
+
+            Binding {
+                target: root
+                property: "title"
+                value: view.name
+            }
+
+            Instantiator {
+                id: instantiator
+
+                model: SortFilterProxyModel {
+                    sourceModel: root.tokensModel
+                    filters: IndexFilter {
+                        minimumIndex: view.index
+                        maximumIndex: view.index
+                    }
+                }
+                delegate: QtObject {
+                    readonly property int deployState: model.deployState
+                    readonly property string name: model.name
+                    readonly property string artworkSource: model.image
+                    readonly property string symbol: model.symbol
+                    readonly property string description: model.description
+                    readonly property int supplyAmount: model.supply
+                    readonly property bool infiniteSupply: model.infiniteSupply
+                    readonly property bool transferable: model.transferable
+                    readonly property bool selfDestruct: model.remoteSelfDestruct
+                    readonly property int chainId: model.chainId
+                    readonly property string chainName: model.chainName
+                    readonly property string chainIcon: model.chainIcon
+                    readonly property string accountName: model.accountName
```

Comment:
I was thinking how to improve that part to avoid exposing model stuff here via properties in order to use it later in `view` as `instantiator.object ? instantiator.object.*: ""`. First I wanted to add some component handling that, but finally I concluded that even if more concise it would be harder to read bc of quite unusual api.

What I finally came up is as follows:
```
            Instantiator {
                model: ...

                delegate: QtObject {
                    component Bind: Binding { target: view }

                    readonly property list<Binding> bindings: [
                        Bind { property: "deployState"; value: model.deployState },
                        Bind { property: "name"; value: model.name },
                        Bind { property: "artworkSource"; value: model.image }
                        // ...
                    ]
```
Than we no longer need to refer to instantiator.object in the `view` object.
The whole part `deployState: instantiator.object ? ...` can be removed.

It's shorter, the binding is defined in a single place and the intention what we are doing here is still quite obvious for the reader. Wdyt?

Btw, I'm waiting for Storybook page for that in order to test this proposition on my own :)

Reply:
That's a good idea, yes :)

Reply:
> 

`Storybook` support added!! I can also try your proposition!! Thanks for thinking on that improvement!